crates/red_knot/src/types/infer.rs

```diff
@@ -80,7 +80,18 @@ where
                     .resolve(ast.as_any_node_ref())
                     .expect("node key should resolve");
 
-                let ty = type_store.add_function(file_id, &node.name.id).into();
+                let decorator_tys: Vec<_> = node
+                    .decorator_list
+                    .iter()
+                    .map(|decorator| {
+                        infer_expr_type(db, file_id, &decorator.expression)
```

Comment:
not sure of correct `file_id` here; should probably try to obtain one from the decorator expression somehow?

Reply:
It's meant to be the file_id in which the expr exists (it's used for calling back into `infer_symbol_type` to resolve names in the expression), so this is correct!

Reply:
Great. Thanks.